src/library/scala/collection/immutable/ListMap.scala

```diff
@@ -127,11 +127,23 @@ sealed class ListMap[K, +V]
       else containsInternal(cur.next, k)
 
     override def updated[V2 >: V1](k: K, v: V2): ListMap[K, V2] = {
-      val m = this - k
-      new m.Node[V2](k, v)
+      replaceInternal(k, v, this, Nil)
     }
 
-    override def remove(k: K): ListMap[K, V1] = removeInternal(k, this, Nil)
+    @tailrec private[this] def replaceInternal[V2 >: V1](k: K, v : V2, cur: ListMap[K, V2], acc: List[ListMap[K, V2]]): ListMap[K, V2] =
+      if (cur.isEmpty) {
+        val last = acc.last
+        new last.Node(k,v)
```

Comment:
can you just return `new this.Node(k, v)` here? Won't `acc.last` always be `this`?

Reply:
@joshlemer you are right. I have done the change and pushed